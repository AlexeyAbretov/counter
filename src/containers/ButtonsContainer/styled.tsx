import styled from 'styled-components';

export const ButtonsWrapper = styled.div`
  display: flex;

  @media (max-width: 450px) {
    flex-direction: column;
  }
`;
