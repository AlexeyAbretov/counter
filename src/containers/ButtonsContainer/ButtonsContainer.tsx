/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/destructuring-assignment */
import React, { useContext } from 'react';
import { observer } from 'mobx-react-lite';

import { Button } from '@components';

import { CounterContext } from '@store';
import { ButtonsWrapper } from './styled';

export const ButtonsContainer: React.FC = observer((): JSX.Element => {
  const counter = useContext(CounterContext);

  return (
    <ButtonsWrapper>
      <Button
        title="Уменьшить"
        type="decrease"
        onClick={() => counter.decrement()}
      />
      <Button title="Сбросить" type="reset" onClick={() => counter.reset()} />
      <Button
        title="Увеличить"
        type="increase"
        onClick={() => counter.increment()}
      />
    </ButtonsWrapper>
  );
});
