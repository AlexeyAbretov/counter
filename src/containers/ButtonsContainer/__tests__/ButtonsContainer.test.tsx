/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import React from 'react';
import { theme } from '@theme';
import { AnyAction, Store } from '@reduxjs/toolkit';
import { CounterStateType } from '@types';
import configureStore from 'redux-mock-store';
import { InitialCounter } from '@store';
import { fireEvent, render } from '@testing-library/react';
import { ButtonsContainer } from '../ButtonsContainer';

const mockStore = configureStore<CounterStateType>();

const getUi = (store: Store<CounterStateType, AnyAction>): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <ButtonsContainer />
      </Provider>
    </ThemeProvider>
  );
};

describe('ButtonsContainer', () => {
  it('should render', () => {
    const { container } = render(
      getUi(
        mockStore({
          ...InitialCounter,
        }),
      ),
    );

    expect(container).toMatchSnapshot();
  });

  it('should call decrement action', () => {
    const store = mockStore({
      ...InitialCounter,
    });

    const { getByText } = render(getUi(store));

    const button = getByText('Уменьшить');

    fireEvent.click(button);

    expect(store.getActions()).toMatchSnapshot();
  });
});
