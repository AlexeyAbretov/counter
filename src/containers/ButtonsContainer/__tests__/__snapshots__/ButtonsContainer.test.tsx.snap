// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsContainer should render 1`] = `
.c1 {
  border: 1px solid aqua;
  border-radius: 6px;
  background-color: transparent;
  width: 140px;
  height: 47px;
  cursor: pointer;
  background-color: indianred;
}

.c1:hover {
  opacity: 0.5;
}

.c2 {
  border: 1px solid aqua;
  border-radius: 6px;
  background-color: transparent;
  width: 140px;
  height: 47px;
  cursor: pointer;
  background-color: darkgrey;
}

.c2:hover {
  opacity: 0.5;
}

.c3 {
  border: 1px solid aqua;
  border-radius: 6px;
  background-color: transparent;
  width: 140px;
  height: 47px;
  cursor: pointer;
  background-color: aquamarine;
}

.c3:hover {
  opacity: 0.5;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:450px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
    >
      Уменьшить
    </button>
    <button
      class="c2"
    >
      Сбросить
    </button>
    <button
      class="c3"
    >
      Увеличить
    </button>
  </div>
</div>
`;
